### PR TITLE
Dpp 1274 view transfer cohorts summary

### PR DESCRIPTION
**Pull request by Pauljgraham**

### CODE REVIEW

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/Views/Commitment/TransferFunded.cshtml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        </table>
+    </div>
+</div>
+
+
+@section breadcrumb {
+    <div class="breadcrumbs">
+        <a href="@Url.Action("Cohorts")" aria-label="Back to your cohorts" class="back-link">Back to your cohorts</a>
+    </div>
+}
```
Comment (lazcool) [inline src/SFA.DAS.ProviderApprenticeshipsService.Web/Views/Commitment/TransferFunded.cshtml]: Interesting...
I was looking for the Google Analytics virtual page view that the employer's web site views have, but the provider's web site views don't have them.

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/Views/Commitment/TransferFunded.cshtml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                        @commitment.ReceivingEmployerName &nbsp;
+                    </td>
+                    <td>
+                        @commitment.HashedCommitmentId &nbsp;
+                    </td>
+                    <td>
+                        @commitment.Status &nbsp;
+                    </td>
+                    <td>
+                        <a href="@Url.Action("Details", new {Model.ProviderId, commitment.HashedCommitmentId})" title="View  @commitment.ReceivingEmployerName" aria-label="View  @commitment.ReceivingEmployerName">Details</a>
```
Comment (lazcool) [inline src/SFA.DAS.ProviderApprenticeshipsService.Web/Views/Commitment/TransferFunded.cshtml]: looks like 2 spaces between View and @commitment... in the title.

Comment (lazcool) [reply src/SFA.DAS.ProviderApprenticeshipsService.Web/Views/Commitment/TransferFunded.cshtml]: ... and the aria-label

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/Views/Commitment/TransferFunded.cshtml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            {
+                <tr>
+                    <td>
+                        @commitment.ReceivingEmployerName &nbsp;
+                    </td>
+                    <td>
+                        @commitment.HashedCommitmentId &nbsp;
+                    </td>
+                    <td>
+                        @commitment.Status &nbsp;
```
Comment (lazcool) [inline src/SFA.DAS.ProviderApprenticeshipsService.Web/Views/Commitment/TransferFunded.cshtml]: I don't think these &nbsp; are required.
In the employer's version of this page, they were in the source page I copied from, but I removed them and everything still seemed hunky dory.

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/Orchestrators/CommitmentOrchestrator.cs [hunk 4]
```diff
@@ -304,6 +317,18 @@ public class CommitmentOrchestrator : BaseCommitmentOrchestrator
             return data.Commitments.Where(m => m.GetStatus() == requestStatus);
         }
 
+        public async Task<IEnumerable<CommitmentListItem>> GetAllCommitmentsWithTheseStatuses(long providerId, RequestStatus[] requestStatus)
```
Comment (lazcool) [inline src/SFA.DAS.ProviderApprenticeshipsService.Web/Orchestrators/CommitmentOrchestrator.cs]: You could use params here to make the calling code look a little better.

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/Views/Commitment/TransferFunded.cshtml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            }
+            </tbody>
+        </table>
+    </div>
+</div>
+
+
+@section breadcrumb {
+    <div class="breadcrumbs">
+        <a href="@Url.Action("Cohorts")" aria-label="Back to your cohorts" class="back-link">Back to your cohorts</a>
```
Comment (lazcool) [inline src/SFA.DAS.ProviderApprenticeshipsService.Web/Views/Commitment/TransferFunded.cshtml]: The mockup in the story has the breadcrumb reading just 'Back', but I think 'Back to your cohorts' is correct as that matches the other sibling pages supplied by RequestList.cshtml

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/Orchestrators/CommitmentOrchestrator.cs [hunk 6]
```diff
@@ -140,6 +140,19 @@ public class CommitmentOrchestrator : BaseCommitmentOrchestrator
             };
         }
 
+        public async Task<TransferFundedViewModel> GetAllTransferFunded(long providerId)
+        {
+            var data = (await GetAllCommitmentsWithTheseStatuses(providerId, new [] { RequestStatus.WithSenderForApproval, RequestStatus.RejectedBySender })).ToList();
+            _logger.Info($"Provider getting all transfer funded ({data.Count}) :{providerId}", providerId);
```
Comment (lazcool) [inline src/SFA.DAS.ProviderApprenticeshipsService.Web/Orchestrators/CommitmentOrchestrator.cs]: The log message could be improved to use the correct tense, as they have already been fetched, e.g. "Got {data.Count} transfer funded cohorts for provider {providerId}"

Comment (Pauljgraham) [reply src/SFA.DAS.ProviderApprenticeshipsService.Web/Orchestrators/CommitmentOrchestrator.cs]: I'll leave in the existing format. AS it's in this format for teh other similar methods

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/Views/Commitment/TransferFunded.cshtml (hunk 5 above)
Comment (Pauljgraham) [reply src/SFA.DAS.ProviderApprenticeshipsService.Web/Views/Commitment/TransferFunded.cshtml]: done

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/Orchestrators/CommitmentOrchestrator.cs (hunk 4 above)
Comment (Pauljgraham) [reply src/SFA.DAS.ProviderApprenticeshipsService.Web/Orchestrators/CommitmentOrchestrator.cs]: done

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/Views/Commitment/TransferFunded.cshtml (hunk 3 above)
Comment (Pauljgraham) [reply src/SFA.DAS.ProviderApprenticeshipsService.Web/Views/Commitment/TransferFunded.cshtml]: done

File: src/SFA.DAS.ProviderApprenticeshipsService.Web/Views/Commitment/TransferFunded.cshtml (hunk 2 above)
Comment (Pauljgraham) [reply src/SFA.DAS.ProviderApprenticeshipsService.Web/Views/Commitment/TransferFunded.cshtml]: done

**Pull request merged**